Lib/idlelib/editor.py

```diff
@@ -1281,14 +1281,20 @@ def smart_backspace_event(self, event):
         # Delete whitespace left, until hitting a real char or closest
         # preceding virtual tab stop.
         chars = text.get("insert linestart", "insert")
+        echars = text.get("insert", "insert lineend")
         if chars == '':
             if text.compare("insert", ">", "1.0"):
                 # easy: delete preceding newline
                 text.delete("insert-1c")
             else:
                 text.bell()     # at start of buffer
             return "break"
-        if  chars[-1] not in " \t":
+        if chars.startswith(self.prompt_last_line):
+            isindent = _line_indent_re.fullmatch(chars[len(self.prompt_last_line):])
+        else:
+            isindent = _line_indent_re.fullmatch(chars)
```

Comment:
prompt_last_line != '' currently signals that we are in Shell input with prompt chars on the first line.  With the prompt removed either to the left or above, this condition will always be false and prompt_last_list will be '' until it is removed.  On the other hand, string.endswith('') will be true, the opposite of what you want.  
```suggestion
           isindent = _line_indent_re.fullmatch(chars)
```
It will not matter in a sense because `string[0:] == string`, making both branches the same, but we do not need redundant branches.